File: array/152.py
#TLE

class Solution:
    def maxProduct(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        
        ret = - sys.maxsize
        
        for i in range(len(nums)):
            j = i + 1
            temp = nums[i]
            ret = temp if temp > ret else ret
            while j < len(nums):
                temp *= nums[j]
                ret = temp if temp > ret else ret
                j += 1
        return ret
 
# not TLE ...               
int maxProduct(int* nums, int numsSize) {
    if(numsSize == 0){
        return 0;
    }
    int ret = INT_MIN;
    int temp;
    for(int i = 0; i < numsSize; i++){
        int j = i + 1;
        temp = nums[i];
        ret = ret > temp ? ret : temp;
        while(j < numsSize){
            temp *= nums[j++];
            ret = ret > temp ? ret : temp;
        }
    }
    return ret;
    
    
}

File: array/152.cc
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        // 乘法跟加法不同, 从初识位置开始的每一个值都应该计入
        int ret = INT_MIN;
        int cur = 1;
        for (int i = 0; i < nums.size(); i++) {
            cur *= nums[i];
            if (cur > ret) {
                ret = cur;
            }
            if (cur == 0) {
                cur = 1;
            }
        }
        cur = 1;
        for (int i = nums.size()-1; i >= 0; i--) {
            cur *= nums[i];
            if (cur > ret) {
                ret = cur;
            }
            if (cur == 0) {
                cur = 1;
            }
        }
        return ret;
    }
};
